### ISSUE TITLE
Improve code and add features.

### ISSUE THREAD

**Issue by artisticLogicMK** [artisticLogicMK/timelydiff#1]
A call to new or experienced developers to make TimelyDiff even better.

**Comment by epreston** [comment]
Hi there, would you be interested in unit tests and example project using vite / vitest ?  

Here's what I feel I can achieve.

- unit tests using vitest for existing functionality and edge cases
- CI configuration for github to trigger the unit tests against each pull request
- test stubs for future / missing functionality
- pull requests for anything I can improve and validate with the tests
- graphical example project, handy tool following the code sample provided

**Comment by artisticLogicMK** [comment]
Hello, thank you for reaching out and expressing your interest in contributing to the code. I appreciate your enthusiasm.

However, this package was built to maintain independence and avoid introducing additional dependencies. The code is designed to be self-contained without relying on external packages.

If you have any suggestions or ideas for improving the existing functionality, feel free to submit any pull requests you think can benefit the project. Thanks again for your interest and contribution. I'm surprised I stumbled upon this after ages.

**Comment by epreston** [comment]
Anytime, happy to help. You got me a little confused.

Tests, documentation, and Github configuration don't add dependencies to the code.

**Comment by artisticLogicMK** [comment]
I might be unaware of or missing something--doesnt vite/vitest require adding it as a development dependency in the package.json file?

However, despite the need for adding this dependencies, I still highly encourage you to proceed with the other aspects you mentioned. You can definitely submit pull requests for any improvements you identify, and also create a graphical example project or a handy tool that demonstrates the usage of the code.

Your contributions in these areas would be valuable and greatly appreciated. Thank you.

**Comment by epreston** [comment]
Yes, a development dependency has nothing to do with the code.  Its purely for development and does not add one byte to what people consume.

The tools used for checking and testing the code need to be defined somewhere.  It's like an example application, it might not be shipped with the module but is there for people who are exploring the repo.

**Comment by artisticLogicMK** [comment]
Please go ahead and proceed with the unit tests using vite/vitest, as well as any other contributions you would like to add. Unit tests will ensure the correctness of the code, and also help other developers learn something new when exploring the repository.

Thank you for your willingness to contribute! 🙏
